src/forms/BoundForm.tsx

```diff
@@ -0,0 +1,237 @@
+  const { inputRows, formState } = props;
+
+  const tid = useTestIds({}, "boundForm");
+
+  return (
+    <div {...tid}>
+      <FormLines labelSuffix={{ required: "*" }} width="full" gap={4}>
+        {inputRows.map((row) => (
+          <FormRow key={`fieldGroup-${Object.keys(row).join("-")}`} row={row} formState={formState} />
+        ))}
+      </FormLines>
+    </div>
+  );
+}
+
+function FormRow<F>({ row, formState }: { row: BoundFormRowInputs<F>; formState: ObjectState<F> }) {
+  const tid = useTestIds({}, "boundFormRow");
+
+  /**  Extract the bound input components with their sizing config or render any "custom" JSX node as-is */
+  const componentsWithConfig = useMemo(() => {
+    return safeEntries(row).map(([key, fieldFnOrCustomNode]) => {
+      if (typeof fieldFnOrCustomNode === "function") {
+        const field = formState[key] ?? fail(`Field ${key.toString()} not found in formState`);
+        const { component, minWith } = fieldFnOrCustomNode(field);
+
+        return { component, key, minWith };
+      }
+
+      return { component: fieldFnOrCustomNode as ReactNode, key };
+    });
+  }, [row, formState]);
+
+  // We can hook into the formState loading state and show a skeleton from that matches the real forms layout
+  const isLoading = useComputed(() => formState.loading, [formState]);
+
+  // Prefer to evenly distribute the available space to each item, but leave some room for the "gap" padding
+  // Then fall back to each item's "min-width" to allow for input-specific sizing when the available space is small
+  const itemFlexBasis = 100 / componentsWithConfig.length - 3;
+
+  return (
+    <div css={Css.df.fww.gap2.$} {...tid}>
+      {componentsWithConfig.map(({ component, key, minWith }) => (
+        <div css={Css.mw(minWith).fb(`${itemFlexBasis}%`).fg1.$} key={key.toString()}>
+          {isLoading ? <LoadingSkeleton size="lg" /> : component}
+        </div>
+      ))}
+    </div>
+  );
+}
+
+/**
+ * These field component functions are thin wrappers around the `BoundFoo` components which omit
+ * certain props that the caller doesn't need to pass or we specifically want to restrict to drive UX consistency.
+ */
+
+// We map the `field` to the bound component automatically for the caller.
+type KeysToOmit = "field";
+
+// Potential TODO: add type overloads for the different HasIdIsh/HasNameIsh combinations, maybe there's a generic way to introspect those types?
+export function boundSelectField<O, V extends Value>(props: Omit<BoundSelectFieldProps<O, V>, KeysToOmit>) {
+  return (field: FieldState<any>): BoundFieldInputFnReturn => ({
+    component: <BoundSelectField field={field} {...props} />,
+    minWith: "200px",
+  });
+}
+
+export function boundMultiSelectField<O, V extends Value>(props: Omit<BoundMultiSelectFieldProps<O, V>, KeysToOmit>) {
+  return (field: FieldState<any>): BoundFieldInputFnReturn => ({
+    component: <BoundMultiSelectField field={field} {...props} />,
+    minWith: "200px",
```

Comment:
Probably will need to add a `maxWidth` as well for the naturally shorter inputs like checkboxes and switches? Though they could have long names so we could optionally just remove the `flex-grow` from them so they just size themselves vs. evenly dividing their width
<img width="711" alt="image" src="https://github.com/user-attachments/assets/32dbcb4d-bd16-4b09-8c81-8fa5df26c78e" />

Reply:
Addressed this by changing the default switch style to `inline` per design feedback as well as adding a `min-content` rule for allowing variable sized inputs to size themselves vs. going with a hardcoded min. 
https://github.com/homebound-team/beam/pull/1122/commits/12ce84de87541deaecafb7f0165a2b0cb798fdd6